### PR TITLE
device: convert temperature and add range

### PR DESCRIPTION
**Pull request by terinjokes**
Update the temperature conversion functions to more closely match the
Elgato Control Center application.

Add assertions to client's SetLights function to verify brightness and
temperatures are within range of all lights.

### CODE REVIEW

File: device.go [hunk 1]
```diff
@@ -12,6 +12,18 @@ import (
 	"time"
 )
 
+const (
```
Comment (mdlayher) [inline device.go:15]: Unexport all of these and change to camel case `brightnessMin` etc and just mention the allowed limits on the field docs within Light.

File: device.go [hunk 2]
```diff
@@ -128,6 +140,16 @@ func (c *Client) Lights(ctx context.Context) ([]*Light, error) {
 
 // SetLights configures the state of all lights on a Key Light device.
 func (c *Client) SetLights(ctx context.Context, lights []*Light) error {
+	for _, light := range lights {
```
Comment (mdlayher) [inline device.go]: Range variable `l` instead of `light`.

File: device.go [hunk 3]
```diff
@@ -188,10 +210,16 @@ func (c *Client) do(ctx context.Context, method, path string, body io.Reader, ou
 
 // convertToKelvin converts the Elgato API temperatures to Kelvin.
 func convertToKelvin(elgato int) int {
-	return int(math.Round(10000*math.Pow(float64(elgato), -1)) * 100)
+	kelvin := TEMP_CONSTANT - int(math.Round(float64(elgato)*TEMP_COEFFICENT))
```
Comment (mdlayher) [inline device.go]: Do you have a formula reference?

File: device_test.go [hunk 4]
```diff
@@ -111,7 +111,7 @@ func TestClientSetLights(t *testing.T) {
 				light,
 				// This light doesn't actually exist and should prompt the client
 				// to return an error.
-				{On: true},
+				{On: true, Temperature: 2900, Brightness: 15},
```
Comment (mdlayher) [inline device_test.go]: At this point just specify `light` in the second slice argument.

File: device_test.go [hunk 5]
```diff
@@ -123,6 +123,28 @@ func TestClientSetLights(t *testing.T) {
...
 			lights: []*keylight.Light{light},
 		},
+		{
+			name: "temperature outside of range",
+			lights: []*keylight.Light{
+				{On: true, Temperature: 2899, Brightness: 15},
+			},
+			check: func(t *testing.T, err error) {
+				if !strings.Contains(err.Error(), "temperature (2899) out of range 2900 <= x <= 7000") {
+					t.Fatalf("error did not mention malformed lights input: %v", err)
```
Comment (mdlayher) [inline device_test.go]: s/lights/temperature/

File: device_test.go [hunk 6]
```diff
@@ -123,6 +123,28 @@ func TestClientSetLights(t *testing.T) {
...
+			},
+		},
+		{
+			name: "brightness outside of range",
+			lights: []*keylight.Light{
+				{On: true, Temperature: 2900, Brightness: 101},
+			},
+			check: func(t *testing.T, err error) {
+				if !strings.Contains(err.Error(), "brightness (101) out of range 3 <= x <= 100") {
+					t.Fatalf("error did not mention malformed lights input: %v", err)
```
Comment (mdlayher) [inline device_test.go]: s/lights/brightness/

File: device.go [hunk 7]
```diff
@@ -76,8 +88,12 @@ type Light struct {
 
 // A jsonLight is the raw JSON representation of a Light.
 type jsonLight struct {
-	On          int `json:"on"`
-	Brightness  int `json:"brightness"`
+	On int `json:"on"`
+
+	// Brigthness has a range of 3-100
```
Comment (mdlayher) [inline device.go]: Apologies, I meant to append these limitations to the doc comments on Light since that is exported for the user to see.

**Review by mdlayher — COMMENTED** [review]
LGTM with docs nit.

File: device.go (hunk 7 above)
Comment (terinjokes) [reply device.go]: 😝

**Review by mdlayher — APPROVED** [review]
LGTM!

**Pull request merged**